realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java

```diff
@@ -134,7 +134,7 @@
      * After being inserted any changes to the original objects will not be persisted.
      *
      * @param realm reference to the {@link Realm} where the objects will be inserted.
-     * @param objects Collection of {@link RealmObject} to insert or update.
+     * @param objects Collection of {@link RealmObject} to insert or update. This must not be empty.
```

Comment:
no, if the `objects` is empty, some mediators actually throws an exception.

see https://github.com/realm/realm-java/pull/3106#discussion-diff-69291749